src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp

```diff
@@ -715,5 +722,304 @@ struct TraceExtrinsicCurvatureCompute
       const tnsr::II<DataVector, SpatialDim, Frame>&) = &trace;
   using base = gr::Tags::TraceExtrinsicCurvature<DataVector>;
 };
+
+// @{
+/*!
+ * \ingroup GeneralizedHarmonicGroup
+ * \brief Compute item to compute constraint-damping parameters for a single-BH
+ * evolution.
+ *
+ * \details Can be retrieved using
+ * `GeneralizedHarmonic::Tags::ConstraintGamma0`,
+ * `GeneralizedHarmonic::Tags::ConstraintGamma0`, and
+ * `GeneralizedHarmonic::Tags::ConstraintGamma0`.
+ */
+template <size_t SpatialDim, typename Frame>
+struct ConstraintGamma0Compute : ConstraintGamma0, db::ComputeTag {
+  using argument_tags = tmpl::list<::Tags::Coordinates<SpatialDim, Frame>>;
+  static auto function(
+      const tnsr::I<DataVector, SpatialDim, Frame>& coords) noexcept {
+    const DataVector r_squared = get(dot_product(coords, coords));
+    Scalar<DataVector> gamma = make_with_value<type>(coords, 0.);
+    get(gamma) = 3. * exp(-0.5 * r_squared / 64.) + 0.001;
+    return gamma;
+  }
+  using base = ConstraintGamma0;
+  using type = typename base::type;
+};
+
+template <size_t SpatialDim, typename Frame>
+struct ConstraintGamma1Compute : ConstraintGamma1, db::ComputeTag {
+  using argument_tags = tmpl::list<::Tags::Coordinates<SpatialDim, Frame>>;
+  static constexpr auto function(
+      const tnsr::I<DataVector, SpatialDim, Frame>& coords) noexcept {
+    return make_with_value<type>(coords, -1.);
+  }
+  using base = ConstraintGamma1;
+  using type = typename base::type;
+};
+
+template <size_t SpatialDim, typename Frame>
+struct ConstraintGamma2Compute : ConstraintGamma2, db::ComputeTag {
+  using argument_tags = tmpl::list<::Tags::Coordinates<SpatialDim, Frame>>;
+  static auto function(
+      const tnsr::I<DataVector, SpatialDim, Frame>& coords) noexcept {
+    const DataVector r_squared = get(dot_product(coords, coords));
+    Scalar<DataVector> gamma = make_with_value<type>(coords, 0.);
+    get(gamma) = exp(-0.5 * r_squared / 64.) + 0.001;
+    return gamma;
+  }
+  using base = ConstraintGamma2;
+  using type = typename base::type;
+};
+// @}
+
+/*!
+ * \ingroup GeneralizedHarmonicGroup
+ * \brief  Compute item to get the implicit gauge source function from 3 + 1
```

Comment:
Okay, this is the only thing left that I don't understand. I'm not making a change request (yet at least :) ). Could you please explain what you mean? :)

Reply:
Sorry for not getting back on this earlier -

The gauge source function `H_a` is supposed to act as a _source_ term in the evolution equations for lapse `N` and shift `N^k`, i.e.:
```
(∂t − N^k ∂_k)N ~ function of {H_0, H_i, N, N^k}, 
(∂t − N^k ∂_k)N^k ~ another function of {H_i, N, N^k, ...}
```
(see Eq.9-10 of [this paper](https://arxiv.org/pdf/gr-qc/0512093.pdf)). However, instead of adding a sourcing function to the box, `GaugeHImplicitFrom3p1QuantitiesCompute` inverts the above two eqns to get back `{H_0, H_i}` from derivatives of lapse, shift, etc (see [its doc](https://spectre-code.org/group__generalrelativitygroup#ga117a90f0a602f755c96bcb686e988052)). In other words, this compute tag computes the gauge source function that is _implied_ by the current state of evolved 3+1 variables.

To put it another way, this compute item only "gets" the current gauge source function's value that is implicit from evolved 3+1 variables, and cannot be used to "set" them. I thought this distinction is important to highlight in the name of this compute tag since it also derives off the simple tag `GaugeH`, which other compute tags, that actually set user-defined values for `H_a` (such as `DampedHarmonicHCompute`), also inherit from.

Reply:
Okay, that makes sense. Thank you for explaining this to me :)